client_side_validation/src/client_side_validation.rs

```diff
@@ -125,6 +125,17 @@ pub mod commit_strategy {
         }
     }
 
+    // Marker trait for automatically implementing the CommitEncode trait
+    //  (with Merklization strategy) for a vector of that given type
+    pub trait VectorCommitEncode {}
```

Comment:
This approach will not work in a well-defined fashion with the `Merklization` strategy from the above: it's not obvious that using merklization strategy will not merkelize while to merkelize you need to impl a separate trait on the type itself.

Reply:
To be honest, I do not understand the reasons you are giving for not accepting this change, but you are the lead developer and must know better, and I respect that.

The fact is that, the way this library is right now, one cannot apply CommitEncode with the Merklization strategy to a vector of any specific type (different than `u8`, `u16`, `u32`, and `u64`).

Reply:
My approach is not to try to fix issues case by case, but have a consistent design on API and type system. If we use a strategy-based derivation, we should not introduce an alternative API without clear distinction.

As for the vectors, I do not see why you can't apply CommitEncoding with merklization for any vector by doing a newtype on it (while the underlying type must also support commit encode). This is a clear way how you can avoid very unpleasant side-effects and bugs like I described in all details in by comment above: merklization is not a property of a type, it is a property of the vector, and for the same type some vectors with that type must be merkelized, and some are not.

Reply:
@claudiosdc I tried some name refactoring + tests in https://github.com/LNP-BP/rust-lnpbp/pull/179. Collection types are working correctly, no new traits are required. Strange, I am not able to tag you or add you as a reviewer (was able to do that like 10 mins ago)